### ISSUE TITLE
Outer loop Unix runs failing with package restore errors

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#6691]
e.g.
http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/outerloop_osx_debug/25/consoleFull
```
20:25:17   [20:29:24.72] Restoring all packages...
20:25:31 EXEC : error : System.Threading.Overlapped 4.0.1-rc3-23901 provides a compile-time reference assembly for System.Threading.Overlapped on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:31 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:33 EXEC : error : System.Security.AccessControl 4.0.0-rc3-23901 provides a compile-time reference assembly for System.Security.AccessControl on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:33 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:35 EXEC : error : Microsoft.Win32.Registry 4.0.0-rc3-23901 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:35 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:35 EXEC : error : System.Security.Cryptography.Csp 4.0.0-rc3-23901 provides a compile-time reference assembly for System.Security.Cryptography.Csp on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:35 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:49 EXEC : error : System.Threading.Overlapped 4.0.1-rc3-23901 provides a compile-time reference assembly for System.Threading.Overlapped on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:49 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:58 EXEC : error : System.Threading.Overlapped 4.0.1-rc3-23901 provides a compile-time reference assembly for System.Threading.Overlapped on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:58 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:58 EXEC : error : System.Threading.Overlapped 4.0.1-rc3-23901 provides a compile-time reference assembly for System.Threading.Overlapped on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:58 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:59 EXEC : error : System.Threading.Overlapped 4.0.1-rc3-23901 provides a compile-time reference assembly for System.Threading.Overlapped on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:25:59 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:00 EXEC : error : System.Threading.Overlapped 4.0.1-rc3-23901 provides a compile-time reference assembly for System.Threading.Overlapped on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:00 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:00 EXEC : error : System.Security.AccessControl 4.0.0-rc3-23901 provides a compile-time reference assembly for System.Security.AccessControl on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:00 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:04 EXEC : error : Microsoft.Win32.Registry 4.0.0-rc3-23901 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:04 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:13 EXEC : error : Microsoft.Win32.Registry 4.0.0-rc3-23901 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:13 EXEC : error : System.Security.AccessControl 4.0.0-rc3-23901 provides a compile-time reference assembly for System.Security.AccessControl on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:13 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:13 EXEC : error : Microsoft.Win32.Registry 4.0.0-rc3-23901 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:13 EXEC : error : System.Security.AccessControl 4.0.0-rc3-23901 provides a compile-time reference assembly for System.Security.AccessControl on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:13 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:14 EXEC : error : Microsoft.Win32.Registry 4.0.0-rc3-23901 provides a compile-time reference assembly for Microsoft.Win32.Registry on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:14 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:15 EXEC : error : System.Runtime 4.0.0 provides a compile-time reference assembly for System.Runtime on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:15 EXEC : error : System.Reflection 4.0.0 provides a compile-time reference assembly for System.Reflection on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:15 EXEC : error : System.IO 4.0.0 provides a compile-time reference assembly for System.IO on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:15 EXEC : error : System.Text.Encoding 4.0.0 provides a compile-time reference assembly for System.Text.Encoding on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:15 EXEC : error : System.Threading.Tasks 4.0.0 provides a compile-time reference assembly for System.Threading.Tasks on DNXCore,Version=v5.0, but there is no run-time assembly compatible with osx.10.11-x64. [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:15 EXEC : error : Some packages are not compatible with DNXCore,Version=v5.0 (osx.10.11-x64). [/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj]
20:26:26 
/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/build.proj(54,5): error MSB3073: The command ""/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/Tools/dotnetcli/bin/dotnet" restore --packages "/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/packages" --source https://dotnet.myget.org/F/dotnet-core/api/v3/index.json --source https://www.nuget.org/api/v2/ "/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/src" "/Users/dotnet-bot/j/workspace/dotnet_corefx/outerloop_osx_debug/pkg"" exited with code -1.
```

**Comment by Priya91** [comment]
@chcosta can you take a look.Thanks!

**Comment by Priya91** [comment]
@dagood this seems to be similar to #6760

**Comment by dagood** [comment]
Yep, this is the same restore compat check issue as 6760, thanks!

**Comment by chcosta** [comment]
Closed as this is being tracked with issue #6760.
